core/src/main/java/com/graphhopper/routing/DefaultWeightingFactory.java

```diff
@@ -75,8 +75,9 @@ public Weighting createWeighting(Profile profile, PMap requestHints, boolean dis
                 throw new IllegalArgumentException("custom weighting requires a CustomProfile but was profile=" + profile.getName());
             CustomModel queryCustomModel = requestHints.getObject(CustomModel.KEY, null);
             CustomProfile customProfile = (CustomProfile) profile;
-            queryCustomModel = queryCustomModel == null ?
-                    customProfile.getCustomModel() : CustomModel.merge(customProfile.getCustomModel(), queryCustomModel);
+            if (queryCustomModel != null)
+                queryCustomModel.check(customProfile.getCustomModel(), encoder.getMaxSpeed());
```

Comment:
What if `customProfile.getCustomModel().findMaxSpeed(encoder.getMaxSpeed()) < encoder.getMaxSpeed())`? Right now I'm not sure if we should use the server-side custom model's max speed as input to `queryCustomModel.check(_, maxSpeed)` instead?

Reply:
Oh, yes. This is correct. All a bit interweaved :)